src/AnonymousUser/AnonymousUserExtensions.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using Microsoft.AspNetCore.Builder;
+
+namespace InsightArchitectures.AnonymousUser
+{
+    /// <summary>
+    /// Extension methods for the ASP NET Core application builder.
+    /// </summary>
+    public static class AnonymousUserExtensions
+    {
+        /// <summary>
+        /// Adds the <see cref="AnonymousUserMiddleware" /> to the middleware pipeline.
+        /// </summary>
+        /// <param name="builder">The application builder object.</param>
+        /// <param name="configure">An action to customise the middleware options.</param>
+        public static IApplicationBuilder UseAnonymousUser(this IApplicationBuilder builder, Action<AnonymousUserOptions> configure = null)
+        {
+            var options = new AnonymousUserOptions();
+
+            configure?.Invoke(options);
+
+            _ = options.ClaimType ?? throw new NullReferenceException($"{nameof(options.ClaimType)} is null. Please provide a claim type name when configuring the middleware.");
```

Comment:
Since you're throwing an exception if `ClaimType` is null, you should make it more explicit for the library user that this value needs to be provided or provide a default value in the options class